### DISCUSSION TITLE
(fsm) Best place to subscribe to external event service

### DISCUSSION THREAD

**Discussion by raviteja83** [statelyai/xstate#4398] [Q&A] [answered]
How can i subscribe to external event service and make transitions in my state machine created using `@xstate/fsm` as invoke is not supported here.

**Comment by davidkpiano** [comment, marked as answer]
By doing it outside:

```ts
const actor = interpret(machine).start();

window.addEventListener('message', ev => {
  // ...
  actor.send(...)
});
```

**Reply by raviteja83** [reply]
yep doing the same. Thanks for the response.
Just checking if there is anyway to do within the machine.

**Reply by davidkpiano** [reply]
In FSM, there is currently no way to do that.
